### DISCUSSION TITLE
Mismatch results in dev with the flag --location

### DISCUSSION THREAD

**Discussion by metacriacoes** [lumeland/lume#424] [Q&A] [answered]
The code:
```javascript
// _config.ts

import lume from "https://deno.land/x/lume@v1.17.4/mod.ts";

const site = lume({
  location: new URL("http://localhost:8888"),
  server: {
    port: 8888
  }
});

console.log(`site.options.location.href: ${site.options.location.href}`);

export default site;
```

Command A:
```bash
deno run --unstable -A https://deno.land/x/lume@v1.17.4/cli.ts -s -d
```

Output A:
```bash
...
site.options.location.href: http://localhost:3000/
...
```

Command B:
```bash
deno run --unstable -A https://deno.land/x/lume@v1.17.4/cli.ts -s -d --location=http://localhost:8888
```

Output B:
```bash
...
site.options.location.href: http://localhost:8888/
...
```
---
Shouldn't the `--location` flag be just an override of the `location` option? Why the different results?

Thanks.

**Comment by oscarotero** [comment]
If an explicit `--location` is passed to the CLI, it's supposed that you want to override the default `localhost` domain. For example, to use a different hostname (like `hostname.local`).

**Reply by metacriacoes** [reply]
Yes. And if the `location` option is set in the `_config.ts` file, like in the example. Shouldn't the output be the same?

**Reply by oscarotero** [reply, marked as answer]
In this case, the hostname is `localhost`.
You can see the implementation here: https://github.com/lumeland/lume/blob/master/mod.ts#L82-L86
(`options.location` is the value of --location) and `options.serve` is true if the flag `-s` or `--serve` is set.

**Reply by metacriacoes** [reply]
Oh, got it. With `-s`  the `location` is _always_ overrided. Thanks.

Great work! I'm enjoying Lume very much.
